channels/audin/server/audin.c

```diff
@@ -142,16 +142,30 @@ static UINT audin_server_recv_formats(audin_server_context* context, wStream* s,
 		WLog_Print(audin->log, WLOG_WARN,
 		           "cbSizeFormatsPacket is invalid! Expected: %u Got: %zu. Fixing size",
 		           pdu.cbSizeFormatsPacket, Stream_GetPosition(s));
-		pdu.cbSizeFormatsPacket = Stream_GetPosition(s);
+		const size_t pos = Stream_GetPosition(s);
+		if (pos > UINT32_MAX)
+		{
+			error = ERROR_INVALID_PARAMETER;
+			goto fail;
+		}
+		pdu.cbSizeFormatsPacket = (UINT32)pos;
 	}
 
-	pdu.ExtraDataSize = Stream_GetRemainingLength(s);
+	const size_t rem = Stream_GetRemainingLength(s);
+
+	if (rem > UINT32_MAX)
```

Comment:
That is not correct. The size of the `ExtraData` field is arbitrary. It is not capped by some limit.